src/test/schema-form/form-test.js

```diff
@@ -506,12 +506,15 @@ describe('Form', () => {
           schema,
         });
 
-        const input = node.querySelector('input[type=text]');
+        const input = node.querySelector('input[type=number]');
         input.value = 'not a number';
         fireEvent(input, 'change');
 
         expect(node.querySelector('.error-item').textContent.trim()).toEqual(
-          '.field1 should be number',
+          // TODO: uncomment this when jsdom's validation is no longer broken.
+          // it doesn't set input.validity.badInput when it should.
```

Comment:
Is there an upstream issue for this? (or a workaround for displaying the correct error message)

Reply:
No, and no. The `validity` object is read-only, so I couldn’t manually set the proper flags.